centreon/src/Core/Domain/Configuration/User/Model/NewUser.php

```diff
@@ -218,4 +224,44 @@ public function setActivate(bool $isActivate): self
 
         return $this;
     }
+
+    /**
+     * @return string
+     */
+    public function getUserInterfaceViewMode(): string
+    {
+        return $this->userInterfaceViewMode;
+    }
+
+    /**
+     * @param string $userInterfaceViewMode
+     * @return self
+     * @throws \Assert\AssertionFailedException
+     * @throws \InvalidArgumentException
+     */
+    public function setUserInterfaceViewMode(string $userInterfaceViewMode): self
+    {
+        Assertion::minLength(
+            $userInterfaceViewMode,
+            self::MIN_USER_INTERFACE_VIEW_MODE_LENGTH,
+            'User::userInterfaceViewMode'
+        );
```

Comment:
I let you consider using `Assertion::notEmptyString()` insteadof `Assertion::minLength()` for a better understanding.